src/coordinator/tsdb/remote/compressed_codecs.go

```diff
@@ -93,11 +94,28 @@ func compressedSegmentsFromReaders(readers xio.ReaderSliceOfSlicesIterator) (*rp
 	return segments, nil
 }
 
-func compressedTagsFromTagIterator(tagIter ident.TagIterator) ([]*rpc.CompressedTag, error) {
-	tags := make([]*rpc.CompressedTag, 0, tagIter.Remaining())
+func compressedTagsFromTagIteratorWithEncoder(tagIter ident.TagIterator, iterPools encoding.IteratorPools) ([]byte, bool) {
+	if iterPools == nil {
+		return nil, false
+	}
+	encoder := iterPools.TagEncoder().Get()
+	err := encoder.Encode(tagIter)
+	if err != nil {
+		return nil, false
+	}
+	defer encoder.Finalize()
+	data, encoded := encoder.Data()
+	if !encoded {
+		return nil, false
+	}
+	return data.Bytes(), true
+}
+
+func tagsFromTagIterator(tagIter ident.TagIterator) ([]*rpc.Tag, error) {
+	tags := make([]*rpc.Tag, 0, tagIter.Remaining())
```

Comment:
avoid pointer slices

Reply:
Required to conform to what protobuf expects

Reply:
Would we ever need to use this func elsewhere? Might be worth it to export it?

Reply:
Unlikely, as this casts directly to the rpc.Tag type which should only be used for cross DC stuff